src/rtapi/rtapi_compat.c

```diff
@@ -696,11 +706,16 @@ int rtapi_get_tags(const char *mod_name)
 	    perror("cant get  RTLIB_DIR ?\n");
 	    return -1;
 	}
-	strcat(modpath,"/");
-	strcat(modpath, flavor->name);
-	strcat(modpath,"/");
-	strcat(modpath,mod_name);
-	strcat(modpath, flavor->mod_ext);
+	if((strlen(modpath) + 1) < PATH_MAX)
```

Comment:
I didn't do that because 
a) it is not a critical error, the output will just be truncated and 
b) the return value is not an error code, it is potentially used by another function as an actual value, so passing -1 to it could be a very bad idea.